### PR TITLE
Custom server request handler, docs

### LINKED ISSUE #476: API docs
Migrated from #122

Describe:  
- [ ] Extensions adding handlers  
- [x] Custom contents managers, checkpoint managers, etc.  
- [x] Save hooks  
- [ ] Probably some other stuff I'm not thinking of right now

### PR DESCRIPTION
**Pull request by jdfreder**
closes #476

### CODE REVIEW

**Comment by jdfreder** [comment]
@parente and @Carreau I'm using a lot of the code from each of your resources on the topic.  I put some references to your respective sources (links at the bottom), let me know if this is okay or if you'd like me to remove the references.

**Comment by Carreau** [comment]
If we can move content from my side to jupyter docs I'm fine with it. So don't feel obliged to reference me.

File: docs/source/extending/handlers.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+.. code:: python
+
+    from notebook.utils import url_path_join
+    from notebook.base.handlers import IPythonHandler
+
+    class HelloWorldHandler(IPythonHandler):
+        def get(self):
+            self.finish('Hello, world!')
+
+        def load_jupyter_server_extension(nb_server_app):
```
Comment (minrk) [inline docs/source/extending/handlers.rst]: This should be defined outside the class

**Comment by parente** [comment]
Oh, hey, cool. Glad to see the blog post was worth something. Pull whatever you want from it. The ref is fine. I'll update my post to point to the official docs on the topic too.

File: docs/source/extending/handlers.rst (hunk 1 above)
Comment (jdfreder) [inline docs/source/extending/handlers.rst]: Whoops!  Thanks for finding this

Comment (jdfreder) [inline docs/source/extending/handlers.rst]: :heavy_check_mark:

**Comment by damianavila** [comment]
LGTM

**Pull request merged**